aom/src/main/java/com/nedap/archie/adl14/PreviousConversionApplier.java

```diff
@@ -154,17 +157,49 @@ public void removeCreatedUnusedTermCodesAndValueSets() {
         if(this.conversionLog == null) {
             return;
         }
+        removeUnusedValuesets();
+        removeUnusedValues();
+    }
+
+    private void removeUnusedValuesets() {
         Set<String> usedValueSets = gatherUsedValueSets(archetype.getDefinition());
         for(Map.Entry<String, ValueSet> valueSetEntry:this.conversionLog.getCreatedValueSets().entrySet()) {
             if(!usedValueSets.contains(valueSetEntry.getKey())) {
                 //ok, unused, remove it
                 archetype.getTerminology().getValueSets().remove(valueSetEntry.getKey());
-                for(String language:archetype.getTerminology().getTermDefinitions().keySet()) {
-                    archetype.getTerminology().getTermDefinitions().get(language).remove(valueSetEntry.getKey());
+                removeTerminologyEntry(valueSetEntry.getKey());
+            }
+        }
+    }
+
+    private void removeUnusedValues() {
+        Set<String> usedCodes = archetype.getAllUsedCodes();
+        for(Map.Entry<String, CreatedCode> valueEntry:this.conversionLog.getCreatedCodes().entrySet()) {
+            if(valueEntry.getValue().getReasonForCreation() == ReasonForCodeCreation.CREATED_VALUE_FOR_EXTERNAL_TERM
+                    && !usedCodes.contains(valueEntry.getValue().getGeneratedCode())) {
+                //unused code found. Remove!
+                removeTerminologyEntry(valueEntry.getValue().getGeneratedCode());
+            }
+        }
+    }
+
+    private void removeTerminologyEntry(String code) {
+        for(String language:archetype.getTerminology().getTermDefinitions().keySet()) {
+            archetype.getTerminology().getTermDefinitions().get(language).remove(code);
+        }
+        List<String> termCodeBindingsToRemove = new ArrayList<>();
+        if(archetype.getTerminology().getTermBindings() != null) {
+            for (String terminologyId: archetype.getTerminology().getTermBindings().keySet()) {
+                Map<String, URI> termBindings = archetype.getTerminology().getTermBindings().get(terminologyId);
+                termBindings.remove(code);
+                if(termBindings.isEmpty()) {
+                    termCodeBindingsToRemove.add(terminologyId);
                 }
             }
         }
-        //TODO: Gather all unused term bindings and remove from archetype as well
+        for(String terminologyId:termCodeBindingsToRemove) {
+            archetype.getTerminology().getTermBindings().remove(terminologyId);
+        }
```

Comment:
I would also place line 190 and 200-202 inside the if-statement. It makes it more clear that these belong together.